generate-map.py

```diff
@@ -3,38 +3,84 @@
 import sqlite3
 import json
 from geopy.geocoders import Nominatim
+import os
 
 
 def getUserInformation():
-    geolocator = Nominatim(timeout=10, user_agent='Opencast map generator')
+
+    locations = []
 
     cur = sqlite3.connect('user.db').cursor()
     cur.execute('select distinct country, city, organization from user')
     for country, city, organization in cur.fetchall():
-        location = geolocator.geocode({'country': country, 'city': city},
-                                      addressdetails=True) \
-                   or geolocator.geocode('%s, %s' % (country, city),
-                                         addressdetails=True)
-        print((country, city), '->', location)
-        if location:
-            yield (location.latitude, location.longitude, organization)
+        if organization != "None":
+            locations.append(compareCache(country, city, organization))
+    return locations
+
+
+def getGeoCode(country, city, organization, dataList):
+
+    geolocator = Nominatim(timeout=10, user_agent='Opencast map generator')
+    newLocation = geolocator.geocode(country, city, addressdetails=True) \
+        or geolocator.geocode('%s, %s' % (country, city), addressdetails=True)
+    if newLocation:
+        os.remove("cache.json")
+        f = open("cache.json", "a")
```

Comment:
Why remove the file and then append the new content to the non-existing file? Instead, you could just overwrite the file:

```python
open("cache.json", "w")  # or wb
```

Additionally, you never close the file handler. You can close it manually or preferrably use:

```python
with open(…) as f:
    f.write(…)
```

Reply:
Thanks for the suggestions, I will change it tomorrow when I´m back in the office.